lib/ruby/truffle/truffle/bundler-workarounds.rb

```diff
@@ -0,0 +1,582 @@
+    VERIFY_PEER = 1
+  end
+
+  verify_none = <<-HEREDOC
+      ==========================================
+      Workaround: Stub OpenSSL::SSL::VERIFY_NONE
+      Error:
+        uninitialized constant OpenSSL::SSL::VERIFY_NONE
+      Called here:
+        bundler/vendor/net/http/persistent.rb:1142
+  HEREDOC
+  puts verify_none unless $quiet_workaround
+  module SSL
+    VERIFY_NONE = 0
+  end
+
+  #   module Digest
+  #     def self.new(enc)
+  #
+  #     end
+  #   end
+
+  module X509
+    class Store
+      def set_default_paths
+
+      end
+      def add_file(f)
+
+      end
+    end
+
+  end
+
+end
+
+bundler_fetcher = <<-HEREDOC
+==========================================
+Workaround: Use curl in bundler downloader for https requests
+HEREDOC
+puts bundler_fetcher unless $quiet_workaround
+
+require "openssl-stubs"
+
+if bundler_loaded
+  require "bundler/fetcher/downloader"
+
+  module Bundler
+    class Fetcher
+      class Downloader
+        def fetch(uri, options = {}, counter = 0)
+          raise HTTPError, "Too many redirects" if counter >= redirect_limit
+
+          # WORKAROUND START
+          response = if uri.scheme == "https"
+                       resp = CurlResponse.new("1.1", 200, "OK")
+                       resp_raw = `curl -i #{uri}`
+                       blank_line_idx = resp_raw.index("\r\n\r\n")
+                       header = resp_raw[0, blank_line_idx]
+                       resp.body = resp_raw[(blank_line_idx+4)..-1]
+                       if m = /ETag: (\"[[:alnum:]]*\")/.match(header)
+                         resp["ETag"] = m[1]
+                       end
+                       resp
+                     else
+                       response = request(uri, options)
+                     end
+          # WORKAROUND END
+
+          Bundler.ui.debug("HTTP #{response.code} #{response.message}")
+
+          case response
+            when Net::HTTPSuccess, Net::HTTPNotModified
+              response
+            when Net::HTTPRedirection
+              new_uri = URI.parse(response["location"])
+              if new_uri.host == uri.host
+                new_uri.user = uri.user
+                new_uri.password = uri.password
+              end
+              fetch(new_uri, options, counter + 1)
+            when Net::HTTPRequestEntityTooLarge
+              raise FallbackError, response.body
+            when Net::HTTPUnauthorized
+              raise AuthenticationRequiredError, uri.host
+            when Net::HTTPNotFound
+              raise FallbackError, "Net::HTTPNotFound"
+            else
+              raise HTTPError, "#{response.class}#{": #{response.body}" unless response.body.empty?}"
+          end
+        end
+
+      end
+    end
+  end
+end
+
+resolver_search_for = <<-HEREDOC
+==========================================
+Workaround: Change =~ to == for resolver#search_for
+Error:  type mismatch: String given
+        stdlib/rubygems/resolver.rb:237:in `block in search_for'
```

Comment:
I didn't fully figure out how this works but I think it originates from `Gem::Platform` in `rubygems/platform`.

It looks like `Gem::Platform.local` can return a `String` or `Gem::Platform` in some cases. If it returned a `Gem::Platform` then it has it's own `Gem::Platform#=~` implementation to use.